service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -67,9 +78,65 @@
     @Mock
     private GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
 
+    @Mock
+    private C2UploadedEmailContentProvider c2UploadedEmailContentProvider;
+
+    @Mock
+    private IdamApi idamApi;
+
     @InjectMocks
     private NotificationHandler notificationHandler;
 
+    @Nested
+    class C2UploadedNotificationChecks {
+        final String subjectLine = "Lastname, SACCCCCCCC5676576567";
+        final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+            .put("subjectLine", subjectLine)
+            .put("hearingDetailsCallout", subjectLine)
+            .put("reference", "12345")
+            .put("caseUrl", "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
+            .build();
+
+        @BeforeEach
+        void before() throws IOException {
+            given(localAuthorityNameLookupConfiguration.getLocalAuthorityName(LOCAL_AUTHORITY_CODE))
+                .willReturn("Example Local Authority");
+
+            given(c2UploadedEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
+                .willReturn(parameters);
+        }
+
+        @Test
+        void shouldNotNotifyHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+            given(idamApi.retrieveUserDetails(AUTH_TOKEN))
+                .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                    "Hmcts", "Test", HMCTS_ADMIN.getRoles()));
+
+            notificationHandler.sendNotificationForC2Upload(new C2UploadedEvent(callbackRequest(), AUTH_TOKEN, USER_ID));
+
+            SendEmailResponse response = verify(notificationClient, times(0))
+                .sendEmail(eq(C2_UPLOAD_NOTIFICATION_TEMPLATE), eq("hmcts-admin@test.com"),
+                    eq(parameters), eq("12345"));
+
+            assertThat(response).isNull();
```

Comment:
I would prefer an assertion here that notificationClient is called never() I think

Reply:
updated

Reply:
@Gogetter I still see assertThat(response).isNull();